test/parallel/test-dgram-multicast-set-interface.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const dgram = require('dgram');
+
+{
+  const socket = dgram.createSocket('udp4');
+
+  socket.bind(0);
+  socket.on('listening', common.mustCall(() => {
+
+    // Explicitly request default system selection
+    socket.setMulticastInterface('0.0.0.0');
+
+    socket.close();
+  }));
+}
+
+{
+  const socket = dgram.createSocket('udp4');
+
+  socket.bind(0);
+  socket.on('listening', common.mustCall(() => {
+
+    socket.close(common.mustCall(() => {
+      assert.throws(() => { socket.setMulticastInterface('0.0.0.0'); },
+                    /Not running/);
+    }));
+  }));
+}
+
+{
+  const socket = dgram.createSocket('udp4');
+
+  socket.bind(0);
+  socket.on('listening', common.mustCall(() => {
+
+    // Try to set with an invalid interfaceAddress (wrong address class)
+    assert.throws(common.mustCall(() => {
+      socket.setMulticastInterface('::');
+    }));
```

Comment:
In this case, libuv currently makes the call with the other address family so each OS could pick a slightly different error from among any unsupported/unavailable/general ones, (and libuv could also later refactor this into a EINVAL or other error instead of making the call.)

I used this test since I think it is worth the risk to determine that no OS with a little too much compatibility code tries to warp this into a success, but this is one of the errors where the exact response could vary quite a bit (which I added to the docs).